### PR TITLE
 [react-phone-number-input] - Update phone-input props

### PR DESCRIPTION
**Pull request by jlismore**
Replace oudated props with updated forms
Add jsdoc help to most functions
Update util function types

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/catamphetamine/react-phone-number-input>> <<https://catamphetamine.github.io/react-phone-number-input/docs/styleguide/index.html#phoneinputwithcountry>>

### CODE REVIEW

**Comment by AdrienEtienne** [comment]
Thank you @jlismore looks good to me.
Just one comment, should we move the old code into a v1 folder?
And keep your code as a v2?

I know its a quite new package but still there is breaking changes

**Review by AdrienEtienne — APPROVED** [review]
Thank you @jlismore looks good to me.

**Comment by lucksp** [comment]
can the `international` prop be added as per `react-phone-number-input` 
> international: boolean? — If country is specified and international property is true then the phone number can only be input in "international" format for that country, but without "country calling code" part. For example, if country is "US" and international property is not passed then the phone number can only be input in the "national" format for US ((213) 373-4253). But if country is "US" and international property is true then the phone number will be input in the "international" format for US (213 373 4253) without "country calling code" part (+1). This could be used for implementing phone number input components that show "country calling code" part before the input field and then the user can fill in the rest of their phone number in the input field.

File: types/react-phone-number-input/index.d.ts [hunk 1]
```diff
@@ -1,77 +1,185 @@
...
     error?: string;
-    ext?: React.ReactElement;
-    flagComponent?: React.Component<{ country: string, flagsPath: string, flags: FlagsMap, }, object>;
+    flagComponent?: React.ComponentType<{ country: string; flagsPath: string; flags: FlagsMap }>;
     flags?: FlagsMap;
-    flagsPath?: string;
-    getInputClassName?: (params: { disable?: boolean, invalid?: boolean }) => string;
-    id?: string | number;
-    inputComponent?: React.Component<InputComponentProps, object>;
-    international?: boolean;
```
Comment (lucksp) [inline types/react-phone-number-input/index.d.ts]: why was this item removed?

Comment (lucksp) [reply types/react-phone-number-input/index.d.ts]: added back in as part of #42996

**Comment by gcoombe** [comment]
re: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/41957/files#diff-7c250e244beb8218eae22310cb1711b50f59f3c393a925b94be69b307150ab3dR63 I think this should be `InputHTMLAttributes<HTMLInputElement>` not `InputHTMLAttributes<string>`. Otherwise it messes the types up for other event handlers like onBlur etc.

**Pull request merged**